hs/CalendarResponse.hs

```diff
@@ -0,0 +1,336 @@
+module CalendarResponse where
+
+import Data.List (sort, groupBy, sortBy)
+import Data.List.Split (splitOn)
+import Data.Time (Day, addDays, formatTime, getCurrentTime)
+import Happstack.Server (ServerPart, Response, toResponse)
+import Control.Monad.IO.Class (liftIO)
+import System.Locale
+import Database.CourseQueries (returnTutorialTimes, returnLectureTimes)
+import qualified Data.Text as T
+import Text.Read (readMaybe)
+import Database.Tables (Time, timeField)
+import Config (firstMondayFall,
+               lastWednesdayFall,
+               firstMondayWinter,
+               lastMondayWinter,
+               outDay,
+               fallHolidays,
+               winterHolidays)
+
+-- | A string representaion for an ICS file.
+type ICSFile = String
+
+-- | A list of all the events created for all courses.
+type Events = [String]
+
+-- | All the information for a course, including the
+-- time fields obtained from the database and the
+-- information from the cookies.
+type CourseInformation = (TimeFields, Code, Section, Session)
+
+-- | A list of all the time fields for a course as obtained
+-- from the database.
+type TimeFields = [Time]
+
+-- | The code for a course.
+type Code = String
+
+-- | The section for a course.
+type Section = String
+
+-- | The session for a course.
+type Session = String
+
+-- | The current date and time as obtained from the system.
+type SystemTime = String
+
+-- | A list of the information within the time fields
+-- ordered by day.
+type TimeFieldsByDay = [[TimeFieldInfo]]
+
+-- | The information contained in a time Field. The first
+-- Double represents the day and the second one the time.
+type TimeFieldInfo = [Double]
```

Comment:
The organization of some of these types feels a bit strange to read. Do you think that it's possible to clarify some of the types? Are some of these types more unnecessary than others?